sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -232,3 +216,32 @@ case class CheckOverflowInSum(
   override protected def withNewChildInternal(newChild: Expression): CheckOverflowInSum =
     copy(child = newChild)
 }
+
+/**
+ * An add expression which is only used for internal add with decimal type.
+ *
+ * Nota that, `DecimalAdd` does not check overflow which is different with `Add`.
+ */
+case class DecimalAdd(
+    left: Expression,
+    right: Expression,
+    override val dataType: DataType,
+    failOnError: Boolean = SQLConf.get.ansiEnabled) extends DecimalArithmetic {
```

Comment:
where do we use this parameter?

Reply:
it's a `override` filed from `BinaryArithmetic`